src/QsCompiler/CompilationManager/FileContentManager.cs

```diff
@@ -113,6 +118,7 @@ internal FileContentManager(Uri uri, string fileName)
             this.SyncRoot = new ReaderWriterLockSlim(LockRecursionPolicy.SupportsRecursion);
             this.Uri = uri;
             this.FileName = fileName;
+            this.Version = 1;
```

Comment:
Instead of initializing this to 1 and setting it later in OpenFileAsync, can the initial version be passed into the constructor?

Reply:
There are many instances of usage of FileContentManager where there is no associated version. Since this is part of the public API, I'd rather not change the constructor and just limit the usage of this new field to those cases in which we want to record the property value. For this one I'm a bit on the fence, I'd like to hear @bettinaheim 's opinion as well.

The advantage of keeping it outside the constructor is that the change is scoped to only this particular situation, and we don't need to change existing usage of FileContentManager objects outside of the context of interaction with the LSP.

Reply:
One reason I would like to see it as a constructor parameter is that it makes it easier to enforce monotonicity. From the LSP specification, it doesn't seem like the client is prevented from opening a file starting with a negative version number - but from the FileContentManager's perspective, this would look like the version is going from 1 to a negative number, which is not allowed.

Reply:
Given that the server itself does not actually need to do anything with this version and simply stores it temporarily to use it when as part of sending messages back to the client, let's leave it up to the client what it wants to do - independent on what the version is (null, not null, monotonous or not), nothing will negatively impact the server code. Let's keep it nullable, and set/keep the version at null here. We only blindly give back versions we got.

Reply:
> we don't need to change existing usage of FileContentManager objects outside of the context of interaction with the LSP

My perspective is that FileContentManager is not LSP-specific, so it should make sense outside the context of an LSP server. If we are adding a monotonically increasing version number to FileContentManager, then I think we should make sure it's well-defined and integrated with the rest of the class.

In my opinion, that would mean enforcing monotonicity and ensuring that version number is incremented whenever the file contents change. That way, other parts of the code or other programs that use FileContentManager (which may not necessarily know about LSP) can rely on the invariants of the version property.

Reply:
> let's leave it up to the client what it wants to do - independent on what the version is (null, not null, monotonous or not)

I'm not sure why to allow null though - that does not fit the JSON schema provided by the spec? (It seems like the choice of `int?` by the Visual Studio LSP package might be a mistake?)

I believe we currently reject messages that are not `int` or `null`, but if we really want to blindly give back what the client sends us, we should loosen the type to `object` since the client may send us any JSON value...

Reply:
Per @bettinaheim 's comment, the initialized value in the constructor is now `null`. I can't speak for the VS SDK team on their choice of `int?` for `VersionedTextDocumentIdentifier.Version`, but my guess is that it was a special value for absent or non-initialized versions in some contexts. Let's not open it for now to something as wide as `object`, but per Bettina's suggestion, we'll keep it null until it's set by the LSP on a particular file. See commit [76793c67](https://github.com/microsoft/qsharp-compiler/pull/1085/commits/76793c67478d68ee19bfebe70ad8ce31319bbc41) on this.